src/vs/vscode.d.ts

```diff
@@ -1665,6 +1665,16 @@ declare module 'vscode' {
 	}
 
 	/**
+	 * A code action for the [light bulb](https://code.visualstudio.com/docs/editor/editingevolved#_code-action) feature.
+	 */
+	export interface CodeAction extends Command {
+		/**
+		 * Should the lightbulb automatically be shown in the UI?
+		 */
+		dontTriggerLightBulb?: boolean;
```

Comment:
Instead of a boolean, I also considered using a enum for when the lightbulb should be shown or not. For example, base on #33441, it may be expected that the lightbulb is shown when the user has a selection but not when the selection is empty